sklearn/pipeline.py

```diff
@@ -60,10 +60,12 @@ class Pipeline(_BaseComposition):
     The purpose of the pipeline is to assemble several steps that can be
     cross-validated together while setting different parameters. For this, it
     enables setting parameters of the various steps using their names and the
-    parameter name separated by a `'__'`, as in the example below. A step's
-    estimator may be replaced entirely by setting the parameter with its name
-    to another estimator, or a transformer removed by setting it to
-    `'passthrough'` or `None`.
+    parameter name separated by a `'__'`. For instance, if one has a pipeline
```

Comment:
I was thinking to add it in the code snippet in the "Examples" section instead. It avoids to make the docstring longer.

Reply:
Do you mean adding another example or modifying the current one? IMO the power of this functionality largely lies in its use with grid searches, and there is already a nice example in the user guide.

Reply:
We already link to the User Guide. IMO, we can achieve what we want with something like:

```python
...  # previous example
0.88
# An estimator's parameter can be set using `__` syntax.
>>> clf.set_params(C=10.0).fit(X_train, y_train).score(X_test, y_test)
0.76
```

I don't think that we want to have a more complex here or why one would need to set the parameters of a pipeline.

In addition to `FeatureUnion`, could you check if we need a similar change in `make_union` and `make_pipeline`.

Reply:
Sure, will do ASAP.

Reply:
@glemaitre Conversations resolved, please let me know if the examples are still not satisfying.

> In addition to `FeatureUnion`, could you check if we need a similar change in `make_union` and `make_pipeline`.

As for `make_pipeline`, there is a link to `Pipeline` in the first line of docstring and it does not mention `__`, so IMO we don't need to repeat such an example. As for `make_union`, I've made `FeatureUnion` a link as well so that it would be easier to redirect.